liferay-headless/liferay-portal
# LPS-189504: PUT operation Empty relationship node does not reset items related to the entry.

**What is new?**
- Implement `fetchRelatedModel` for Custom Object Defintions in ObjectEntry1MModelsRelated and `System1MModelsRelated`.
- Update 1-M parsers checking valid data structure when it is going to set null data.
- Update `DefaultObjectEntryManager` and `ObjectRelationshipExtensionProvider` to disassociate related object entries in a Many to One relationship.
- Create `fetchManyToOneSystemObjectEntry` in DefaultObjectEntryManager.
- Create some test cases in different scenarios

**How to test this?**

1.Create University object
2. Create Student object
3. Create many-to-many relationship universitiesStudents with University as parent
4. Create University entry and Student entry and relate them. Save universityId for following steps.
5. Execute putUniversity:

```curl
curl -X 'PUT' \
  'http://localhost:8080/o/c/universities/${existingUniversityEntryId}' \
  -H 'accept: application/json' \
  -u '${username}:${password}'
  -d '{
  "name": "Oxford_updated",
  "universitiesStudents": [
  ]
}'
```
`universitiesStudents` relationship should be emptied. 

**How to deploy it?**
- Deploy `object-api` and `object-service`
- Deploy `object-rest-api` and `object-rest-impl`

**JIRA Related Ticket**
https://liferay.atlassian.net/browse/LPS-189504
